### DISCUSSION TITLE
Proxy Subscriptions

### DISCUSSION THREAD

**Discussion by SSaiPrasad7** [emqx/emqx#4556] [Q&A] [answered]
@zmstone 
I have studied and implemented the proxy subscriptions as presented in the Emqx docs but I did not understood usecase of that feature. This is one of the unique features present in Emqx when compared with other servers.

I would love to know the uses and in which situation should I use that particular feature?
Thank you

**Comment by hjianbo** [comment, marked as answer]
There are several situations where this feature is used.

1. old devices are migrated to a new topic format design.
2. when implementing an IoT platform in order to be able to reject all subscription requests from devices. The subscription relationship is all managed by the platform according to the business of the device. (This also results in a security gain. Because the topic format desgin is not leaked)

**Reply by SSaiPrasad7** [reply]
Hello @HJianBo,
Please, Can you elaborate point 1?

**Reply by zmstone** [reply]
this feature in 5.0 will be renamed to `auto-subscribe`
it is basically a set of rules to be applied for certain clients so they get automatically subscribed to topic(s)
so they do not have to send subscribe requests.

> old devices are migrated to a new topic format design.

This is perhaps a use case for topic rewrite ?

**Reply by hjianbo** [reply]
> This is perhaps a use case for topic rewrite ?

However, when the number of subscribed topics is not equal, TopicRewrite is limited

**Reply by SSaiPrasad7** [reply]
@maneetgoyal
